source/agora/node/BlockStorage.d

```diff
@@ -784,13 +784,22 @@ public class BlockStorage : IBlockStorage
 
             size_t record_size = (size_t.sizeof * 2 + Hash.sizeof);
             size_t record_count = idx_file.size / record_size;
+
+            scope DeserializeDg dg = (size) @safe
+            {
+                ubyte[] res;
+                res.length = size;
+                idx_file.rawRead(res);
+                return res;
+            };
+
             foreach (idx; 0 .. record_count)
             {
-                height = idx_file.readSizeType();
+                deserializePart(height, dg, CompactMode.No);
```

Comment:
That's not going to compile, you need to remove the third argument.

Reply:
The compact size will be applied to #354.